src/distbin-html/public.ts

```diff
@@ -13,7 +13,9 @@ import { distbinBodyTemplate } from "./partials"
 
 export const createHandler = ({ apiUrl }: {apiUrl: string}) => {
   return async (req: IncomingMessage, res: ServerResponse) => {
-    res.writeHead(200)
+    res.writeHead(200, {
+      "content-type": "text/html",
+    })
```

Comment:
I think this will only affect the /public URL right now. Might need to add in other handlers as they return HTML.

https://github.com/gobengo/distbin/blob/8fa82ac63a2be38a478143c1bc100a1f304eeb9a/src/distbin-html/index.ts#L13

Reply:
I really don't know, it's far from my knowledge. I discovered node.js and typescript yesterday.

I was creating a [package](https://github.com/yalh76/distbin_ynh) to be able to install distbin on [Yunohost](https://yunohost.org) : one-click installation for newbies

During my tests, the about page didn't render and ask me to download a file named public when going threw Nginx reverse proxy but it was ok going directly. So maybe more an Nginx issue, interpreting a content without content type as application/octet-stream

BTW: I'm very interested by ActivityPub related applications, and you made the only ActivityPub Post/Pastebin application. So very interested for future steps and features of distbin.

Great job ^^

Reply:
@yalh76 Thanks for those nice words, and your idea for code formatting.

The most important next step from my perspective is to do whatever it takes to get it to federate with other ActivityPub apps. e.g.
* In Mastodon, I should be able to 'remote follow' https://distbin.com/public
* I should be able to reply to news on https://prismo.news/ anonymously by posting the reply on distbin

A smaller project would be to pull out the activitypub implementation from here into a reusable module that other projects can use and improve. Did it for activitystreams here https://github.com/gobengo/activitystreams2

Reply:
@yalh76  Did you get it running on Yunohost?

Reply:
Yes, it works. Installation, deleting, backup, restore, Change Url.

For now, I haven't find a way to install it in a path (https://www.example.com/distbin), it has to be installed on the root domain (http://distin.example.com).

But installation works fine. I find a way to define the port and the folder where posts are saved using the systemd service environment https://github.com/YunoHost-Apps/distbin_ynh/blob/master/conf/systemd.service

And as I'm less than a newbie in Typescript, I implemented ts-node to run distbin.

If you have time to loose, you can look at the [bash script](https://github.com/YunoHost-Apps/distbin_ynh/blob/master/scripts/install) that automate the installation and the [systemd service](https://github.com/YunoHost-Apps/distbin_ynh/blob/master/conf/systemd.service) used to run distbin

Yunohost is debian based, (I'm also newbie on linux, so be indulgent ^^)

Reply:
> 
> 
> @yalh76 Thanks for those nice words, and your idea for code formatting.
> 
> The most important next step from my perspective is to do whatever it takes to get it to federate with other ActivityPub apps. e.g.
> 
>     * In Mastodon, I should be able to 'remote follow' https://distbin.com/public
> 
>     * I should be able to reply to news on https://prismo.news/ anonymously by posting the reply on distbin
> 
> 
> A smaller project would be to pull out the activitypub implementation from here into a reusable module that other projects can use and improve. Did it for activitystreams here https://github.com/gobengo/activitystreams2

Yes federation is the key success of your app. The feature I proposed is more a use case: having someone copy/paste a log or something else in distbin and able to make comments from mastodon, etc...

For a Typescript activitypub implementation, maybe you already know, there is [Peertube](https://github.com/Chocobozzz/PeerTube), if it can help ^^